test/unit/update_and_kill_test.lua

```diff
@@ -0,0 +1,97 @@
+    local second_space = helpers.create_space('second_space', cg.params.engine)
+    second_space:create_index('primary')
+    cg.second_space = second_space
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    cg.first_space = helpers.create_space_with_hash_index(cg.params.engine)
+    local second_space = helpers.create_space('second_space', cg.params.engine)
+    second_space:create_index('primary', {type = 'HASH'})
+    cg.second_space = second_space
+end)
+
+g.after_each(function(cg)
+    if cg.task ~= nil then
+        cg.task:kill()
+    end
+    cg.first_space:drop()
+    cg.second_space:drop()
+end)
+
+function g.test_expirationd_update(cg)
+    local first_space = cg.first_space
+    local second_space = cg.second_space
+
+    local first_expd_link = require("expirationd")
+
+    -- Start tasks by first expirationd link.
+    local first_expd_tasks_cnt = 4
+    local first_expd_task_name_prefix = "first_"
+    for i = 1, first_expd_tasks_cnt do
+        first_expd_link.start(first_expd_task_name_prefix .. i, first_space.id, helpers.is_expired_true)
+    end
+
+    -- Check updating in progress message.
+    local chan = fiber.channel(1)
+    fiber.create(function()
+        first_expd_link.update()
+        chan:put(1)
+    end)
+    local _, err = pcall(function() first_expd_link.start() end)
+    t.assert_str_contains(err, "Wait until update is done")
+    chan:get()
+
+    -- Check that links are not equals.
+    local second_expd_link = require("expirationd")
+    t.assert_not_equals(
+        tostring(first_expd_link):match("0x.*"),
+        tostring(second_expd_link):match("0x.*"))
+
+    -- Start tasks by second expirationd link.
+    local second_expd_tasks_cnt = 4
+    local second_expd_task_name_prefix = "second_"
+    for i = 1, second_expd_tasks_cnt do
+        second_expd_link.start(second_expd_task_name_prefix .. i, second_space.id, helpers.is_expired_true)
+    end
+
+    -- Check that we have all tasks be shared between both tasks.
+    t.assert_equals(first_expd_link.tasks(), second_expd_link.tasks())
+
+    -- And tasks work correctly.
+    for _, space in pairs({first_space, second_space}) do
+        local total = 10
+        for i = 1, total do
+            space:insert({i, tostring(i)})
+        end
+
+        t.assert_equals(space:count(), total)
+        helpers.retrying({}, function()
+            t.assert_equals(space:count(), 0)
+        end)
+    end
+
+    for _, task_name in pairs(second_expd_link.tasks()) do
+        second_expd_link.kill(task_name)
+    end
+
+    -- Back old link in require. It's necessary to avoid problem of double update call.
+    -- The problem that we can't use old link properly after double update.
+    -- Old link wouldn't see new tasks which were started by new link. That could happen without this line.
+    -- Expirationd module changes only one previous link that stores in package.loaded.
+    package.loaded["expirationd"] = first_expd_link
```

Comment:
Could you give an example why this is needed?

Reply:
I tried to describe why it's needed in comments above. Double update happens because we run this test three times due to this:
```lua
t.group('update_and_kill', {
    {index_type = 'TREE', engine = 'vinyl'},
    {index_type = 'TREE', engine = 'memtx'},
    {index_type = 'HASH', engine = 'memtx'},
})
```

I don't want to move it in `after_each` because we don't need need this line for further kill test

Reply:
Let's do it then in `after_test`:

https://github.com/tarantool/tarantool/blob/2b1c8713b54f94b0fd3c6e106426b583c06f64ad/test/app-luatest/fiber_slice_test.lua#L20-L22

This will help to return to the original `expirationd` even if the test fails.

Reply:
Done